test/jdk/java/sql/testng/test/sql/TimestampTests.java

```diff
@@ -642,6 +642,35 @@ public void test52(long value, String ts) {
         assertEquals(ts1.toString(), ts, "ts1.toString() != ts");
     }
 
+    @Test
+    public void test53() {
+        // The latest Instant that can be converted to a Timestamp.
+        Instant instant1 = Instant.ofEpochSecond(Long.MAX_VALUE / 1000, 999_999_999);
+        assertEquals(instant1, Timestamp.from(instant1).toInstant());
+
+        // One nanosecond more, and converting it gets an overflow.
+        Instant instant2 = instant1.plusNanos(1);
+        try {
+            Timestamp unused = Timestamp.from(instant2);
+            fail();
+        } catch (IllegalArgumentException expected) {
+        }
+
```

Comment:
TestNG supports a better way for expected exceptions, as an attribute of the @Test annotation. Maybe it can be used here for the expected failing cases?

Reply:
I'm not very familiar with TestNG, as you'll have noticed. But I see it has a better way yet, namely `expectThrows`, like JUnit's `assertThrows`. So I've updated the code to use that. Thanks for the hint!